### PR TITLE
*: merge packages of evaluator and expression.

### PR DESCRIPTION
**Pull request by hanfei1991**
Say good bye to evaluator pkg.
@shenli @zimulala @tiancaiamao @coocood PTAL

### CODE REVIEW

**Comment by shenli** [comment]
LGTM
Please make sure that you could pass tidb-test.

File: expression/expression.go [hunk 1]
```diff
@@ -23,10 +23,21 @@ import (
...
 )
 
+// Error instances.
+var (
+	ErrInvalidOperation = terror.ClassExpression.New(CodeInvalidOperation, "invalid operation")
+)
+
+// Error codes.
+const (
+	CodeInvalidOperation terror.ErrCode = 1
```
Comment (zimulala) [inline expression/expression.go]: s/CodeInvalidOperation/codeInvalidOperation
It only uses in the package of expression. It doesn't need to be exported.

File: expression/expression.go [hunk 2]
```diff
@@ -23,10 +23,21 @@ import (
...
 	"github.com/pingcap/tidb/model"
 	"github.com/pingcap/tidb/mysql"
+	"github.com/pingcap/tidb/terror"
 	"github.com/pingcap/tidb/util/codec"
 	"github.com/pingcap/tidb/util/types"
 )
 
+// Error instances.
+var (
+	ErrInvalidOperation = terror.ClassExpression.New(CodeInvalidOperation, "invalid operation")
```
Comment (zimulala) [inline expression/expression.go]: Ditto.

File: expression/constant_test.go [hunk 3]
```diff
@@ -27,10 +26,6 @@ import (
 	"github.com/pingcap/tidb/util/types"
 )
 
-func TestT(t *testing.T) {
```
Comment (tiancaiamao) [inline expression/constant_test.go]: remove this line will skip all tests !!!

Comment (hanfei1991) [inline expression/constant_test.go]: This line has existed in evaluator test

**Comment by hanfei1991** [comment]
@zimulala  PTAL

**Pull request merged**